tools/caffe_converter/convert_model.py

```diff
@@ -72,9 +72,11 @@ def main():
                 wmat_dim = list(layer_blobs[0].shape)
             wmat = np.array(layer_blobs[0].data).reshape(wmat_dim)
             bias = np.array(layer_blobs[1].data)
-            if first_conv:
-                print 'Swapping BGR of caffe into RGB in mxnet'
-                wmat[:, [0, 2], :, :] = wmat[:, [2, 0], :, :]
+            channels = layer_blobs[0].channels;
+            if channels == 3 or channels == 4: # RGB or RGBA
```

Comment:
Sorry, where is ARGB here? I don't understand what change your suggesting. My main problem here was that it was not accounting for mnist 8-bit gray-scale images. Suggestions?